File: 1344-angle-between-hands-of-a-clock/1344-angle-between-hands-of-a-clock.go
func angleClock(hour int, minutes int) float64 {
    minutesDeg := (float64(minutes) / 5) * 30
    hourDeg := math.Mod((float64(hour) + (float64(minutes) / 60)) * 30, 360)
    deg := abs(minutesDeg - hourDeg)
    if deg >= 180 {
        return 360 - deg
    }
    return deg
}

func abs(n float64) float64 {
    if n < 0 {
        return -n
    }
    return n
}

// math
// time: O(1)
// space: O(1)

File: 1344-angle-between-hands-of-a-clock/1344-angle-between-hands-of-a-clock.py
class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:
        minutesDeg = (minutes / 5) * 30
        hourDeg = ((hour + (minutes / 60)) % 12) * 30
        deg = abs(minutesDeg - hourDeg)
        return deg if deg < 180 else 360 - deg

# math
# time: O(1)
# space: O(1)